### DISCUSSION TITLE
Cannot authenticate trivy operator to private docker registry

### DISCUSSION THREAD

**Discussion by madhavansri2022** [aquasecurity/trivy-operator#1260] [Help & Support]
I followed the docs(https://aquasecurity.github.io/trivy-operator/v0.5.0/tutorials/private-registries/) for authenticating our private registry to scan images. Only the first option seems to work.But we are not comfortable running Trivy as root. The second,third and fourth options seem to get ACCESS DENIED error. I could see that error in the logs.
Logs:
"Error","status.message":"time\t\u001b[31mFATAL\u001b[0m\timage scan error: scan error: unable to initialize a scanner: unable to initialize a docker scanner: 4 errors occurred:\n\t* unable to inspect the image ([registry.private.com/project:4**7](http://registry.private.com/project:4**7)): Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?\n\t* unable to initialize Podman client: no podman socket found: stat podman/podman.sock: no such file or directory\n\t* containerd socket not found: /run/containerd/containerd.sock\n\t* GET https://gitlab.private.com/jwt/auth?scope=repository%3Arepo%2Fproject%3Apull&service=container_registry: DENIED: access forbidden\n\n\n"
Note: I did try above options with the recommended token format.

**Comment by madhavansri2022** [comment]
And also we have a gitlab registry for our images. And we've been following this format for registry authentication.
{"ghcr.io":{"username":"","password":"","email":"","auth":""}}.
Also do we require docker in Trivy operator container to pull images from registries?

**Comment by humpalu** [comment]
I created the docker config with kubectl like described here

https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/

`kubectl create secret docker-registry regcred --docker-server=<your-registry-server> --docker-username=<your-name> --docker-password=<your-pword> --docker-email=<your-email>`

creted In the trivy namespace and defined the secret in the PullSecret block in the Values.yaml

https://github.com/aquasecurity/trivy-operator/blob/1b69d4af7e7289e3f42e17503ab9de851bdb0f66/deploy/helm/values.yaml#L126

It works for me with Azure ACR
